rules/expiring-todo-comments.js

```diff
@@ -0,0 +1,470 @@
+				targetPackageRawEngineVersion
+			);
+
+			const compare = semverComparisonForOperator(engine.condition);
+
+			if (compare(targetPackageEngineVersion, todoEngine)) {
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: MESSAGE_ID_ENGINE_MATCHES,
+					data: {
+						comparison: `${engine.name}${engine.condition}${engine.version}`,
+						message: parseTodoMessage(comment.value)
+					}
+				});
+			}
+		}
+
+		for (const unknown of unknowns) {
+			// In this case, check if there's just an '@' missing before a '>' or '>='.
+			const hasAt = unknown.includes('@');
+			const comparisonIndex = unknown.indexOf('>');
+
+			if (!hasAt && comparisonIndex !== -1) {
+				const testString = unknown.slice(0, comparisonIndex) + '@' + unknown.slice(comparisonIndex);
+
+				if (parseArgument(testString).type !== 'unknowns') {
+					uses++;
+					context.report({
+						node: null,
+						loc: comment.loc,
+						messageId: MESSAGE_ID_MISSING_AT_SYMBOL,
+						data: {
+							original: unknown,
+							fix: testString,
+							message: parseTodoMessage(comment.value)
+						}
+					});
+					continue;
+				}
+			}
+
+			const withoutWhitespaces = unknown.replace(/ /g, '');
+
+			if (parseArgument(withoutWhitespaces).type !== 'unknowns') {
+				uses++;
+				context.report({
+					node: null,
+					loc: comment.loc,
+					messageId: MESSAGE_ID_REMOVE_WHITESPACES,
+					data: {
+						original: unknown,
+						fix: withoutWhitespaces,
+						message: parseTodoMessage(comment.value)
+					}
+				});
+				continue;
+			}
+		}
+
+		return uses === 0;
+	}
+
+	return {
+		Program() {
+			rules.Program(); // eslint-disable-line new-cap
+		}
+	};
+};
+
+const schema = [
+	{
+		type: 'object',
+		properties: {
+			terms: {
+				type: 'array',
+				items: {
+					type: 'string'
+				}
+			},
+			ignoreDatesOnPullRequests: {
+				type: 'boolean'
+			}
+		}
+	}
+];
+
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			url: getDocsUrl(__filename)
+		},
+		messages: {
+			[MESSAGE_ID_AVOID_MULTIPLE_DATES]:
+				'Avoid using multiple expiration dates in TODO: {{expirationDates}}. {{message}}',
+			[MESSAGE_ID_EXPIRED_TODO]:
+				'There is a TODO that is past due date: {{expirationDate}}. {{message}}',
+			[MESSAGE_ID_REACHED_PACKAGE_VERSION]:
+				'There is a TODO that is past due package version: {{comparison}}. {{message}}',
+			[MESSAGE_ID_AVOID_MULTIPLE_PACKAGE_VERSIONS]:
+				'Avoid using multiple package versions in TODO: {{versions}}. {{message}}',
+			[MESSAGE_ID_HAVE_PACKAGE]:
+				'There is a TODO that is deprecated since you installed: {{package}}. {{message}}',
+			[MESSAGE_ID_DONT_HAVE_PACKAGE]:
+				'There is a TODO that is deprecated since you uninstalled: {{package}}. {{message}}',
+			[MESSAGE_ID_VERSION_MATCHES]:
+				'There is a TODO match for package version: {{comparison}}. {{message}}',
+			[MESSAGE_ID_ENGINE_MATCHES]:
+				'There is a TODO match for engine version: {{comparison}}. {{message}}',
```

Comment:
We only support `node`, so we should be more explicit in the message.
```suggestion
				'There is a TODO match for Node.js version: {{comparison}}. {{message}}',
```

Reply:
How about `npm` as engine?

Should we report when we receive something like `iojs` then?

Reply:
> How about npm as engine?

We don't support that right now and I don't really see the point in supporting it at all.

> Should we report when we receive something like iojs then?

No, just ignore it. Nobody is using iojs anymore. Not even worth reporting as invalid.

Reply:
Done